In [1]:
import pandas as pd

In [4]:
df = pd.read_csv("./datasets/covid19_tweets.csv")

In [3]:
df.sample(5)
# mostra 5 linhas aleatorias

user_name             user_location  \
71388   Sir Simon McDonald                       NaN   
120066             Zakiyya              South Africa   
45774            Chris May           Little Rock, AR   
131309        Kyle Delfing  Vernon, British Columbia   
164126        Arizona Bats                       NaN   

                                         user_description  \
71388   Permanent Under-Secretary, Foreign & Commonwea...   
120066  Head of Software Delivery & Operations at a Fi...   
45774                               News anchor @KATVNews   
131309  Entrepreneu/shareholder, #PPC EDA member NOS, ...   
164126  Badass Teachers Association was created to giv...   

               user_created  user_followers  user_friends  user_favourites  \
71388   2015-07-28 08:52:32           29030           177              874   
120066  2009-03-27 17:02:49            1523          2149            24586   
45774   2016-08-01 13:36:41            3967           403              634   
131309  2019-04-18 18:05:00            1184           665             3467   
164126  2015-01-12 00:24:05             845           568             1659   

        user_verified                 date  \
71388            True  2020-08-04 05:18:06   
120066          False  2020-08-13 09:32:27   
45774           False  2020-07-31 18:56:29   
131309          False  2020-08-14 02:14:39   
164126          False  2020-08-22 02:36:00   

                                                     text     hashtags  \
71388   During #Covid19 @foreignoffice switched 900 co...  ['Covid19']   
120066  I love little kids and pets popping into offic...  ['COVID19']   
45774   Lance Taylor: Our goal is to start the high sc...  ['COVID19']   
131309  Walmart - mask\nHair dresser - mask\nChiroprac...          NaN   
164126  We don't need standardized testing. Not in the...  ['Covid19']   

                     source  is_retweet  
71388      Twitter for iPad       False  
120066  Twitter for Android       False  
45774    Twitter for iPhone       False  
131309  Twitter for Android       False  
164126            TweetDeck       False

In [5]:
df["hashtags"].unique()

array([nan, "['COVID19']", "['CoronaVirusUpdates', 'COVID19']", ...,
       "['COVID19', 'coronavirus', 'TrumpVirus']",
       "['COVID19', 'CloseTheSchools', 'KeepTheSchoolsClosed', 'KeepTeachersAlive', 'K12']",
       "['nurses', 'COVID19', 'coronavirus', 'schools']"], dtype=object)

In [6]:
df.shape

(179108, 13)

In [7]:
df.info

<bound method DataFrame.info of                      user_name               user_location  \
0                       ᏉᎥ☻լꂅϮ                  astroworld   
1                Tom Basile 🇺🇸                New York, NY   
2              Time4fisticuffs            Pewee Valley, KY   
3                  ethel mertz        Stuck in the Middle    
4                     DIPR-J&K           Jammu and Kashmir   
...                        ...                         ...   
179103  AJIMATI AbdulRahman O.             Ilorin, Nigeria   
179104                   Jason                     Ontario   
179105             BEEHEMOTH ⏳                   🇨🇦 Canada   
179106           Gary DelPonte               New York City   
179107                 TUKY II  Aliwal North, South Africa   

                                         user_description  \
0       wednesday addams as a disney princess keepin i...   
1       Husband, Father, Columnist & Commentator. Auth...   
2       #Christian #Catholic #Conservati

In [8]:
df.loc[df["user_location"].isna()]

user_name user_location  \
7               Derbyshire LPC           NaN   
8            Prathamesh Bendre           NaN   
10      Voice Of CBSE Students           NaN   
16              ChennaiCityNow           NaN   
18                  Dorian Aur           NaN   
...                        ...           ...   
179083           Auntie Kittie           NaN   
179091             🦋  ROGUE  🦋           NaN   
179092       claudia fernandez           NaN   
179097               rainmaker           NaN   
179098               John Geer           NaN   

                                         user_description  \
7                                                     NaN   
8        A poet, reiki practitioner and a student of law.   
10                                                    NaN   
16      Individual tweeting about significant happenin...   
18                                                    NaN   
...                                                   ...   
179083  I'm the auntie you always wished you had.  If ...   
179091  May all aching hearts find their way there thr...   
179092                                                NaN   
179097                                                NaN   
179098  #StayAtHome #StayAtHomeSaveLifes #MaskUp \nFor...   

               user_created  user_followers  user_friends  user_favourites  \
7       2012-02-03 18:08:10             608           355               95   
8       2015-04-25 08:15:41              25            29               18   
10      2020-07-14 17:50:30               8            10                7   
16      2009-04-26 09:38:11            3987            53              749   
18      2011-01-30 18:40:16              46           108              453   
...                     ...             ...           ...              ...   
179083  2019-12-25 18:36:22             888           545               33   
179091  2017-01-30 07:38:23             628           386            14845   
179092  2015-07-02 23:07:03              68           192             2455   
179097  2017-08-16 22:12:17             709          1158            95006   
179098  2020-04-18 01:55:14              61           168            10817   

        user_verified                 date  \
7               False  2020-07-25 12:27:00   
8               False  2020-07-25 12:26:59   
10              False  2020-07-25 12:26:53   
16              False  2020-07-25 12:26:44   
18              False  2020-07-25 12:26:43   
...               ...                  ...   
179083          False  2020-08-29 19:45:02   
179091          False  2020-08-29 19:44:51   
179092          False  2020-08-29 19:44:49   
179097          False  2020-08-29 19:44:35   
179098          False  2020-08-29 19:44:34   

                                                     text  \
7       You now have to wear face coverings when out s...   
8       Praying for good health and recovery of @Chouh...   
10      49K+ Covid19 cases still no response from \n@c...   
16      July 25 #COVID19 update\n#TamilNadu - 6988\nDi...   
18      It is during our darkest moments that we must ...   
...                                                   ...   
179083  Just a quick note for my lovely members.  Soci...   
179091  @dougmillsnyt @realDonaldTrump @FLOTUS "The #T...   
179092  @politvidchannel The Trump Administration's In...   
179097  @politvidchannel Just not the lives of #COVID1...   
179098  Report #COVID19 outbreaks in K-12 schools here...   

                                                 hashtags  \
7                                                     NaN   
8                            ['covid19', 'covidPositive']   
10                                                    NaN   
16                    ['COVID19', 'TamilNadu', 'chennai']   
18                                              ['light']   
...                                                   ...   
179083                                                NaN   


Nesse momento, podemos perceber que o DataFrame possui valores classificados como NaN

algumas operações podem ser afetadas por tais valores, como plot, manipulação de string...

Portanto, no processo de limpeza de dados, devemos sempre analisar se as demais informações dessas colunas são relevantes, ou se podemos simplesmente remover os dados por completo

In [5]:
df.loc[df["user_description"].isna()]

user_name  user_location user_description  \
7               Derbyshire LPC            NaN              NaN   
10      Voice Of CBSE Students            NaN              NaN   
18                  Dorian Aur            NaN              NaN   
20           The Voice of GenX            NaN              NaN   
30                      Ms Paz  United States              NaN   
...                        ...            ...              ...   
178978               Cody Vest            NaN              NaN   
178988                   Shane        Ireland              NaN   
179037            Carol London            NaN              NaN   
179092       claudia fernandez            NaN              NaN   
179097               rainmaker            NaN              NaN   

               user_created  user_followers  user_friends  user_favourites  \
7       2012-02-03 18:08:10             608           355               95   
10      2020-07-14 17:50:30               8            10                7   
18      2011-01-30 18:40:16              46           108              453   
20      2011-05-28 15:28:21             292          1037               58   
30      2019-09-15 18:10:09             127           974            30217   
...                     ...             ...           ...              ...   
178978  2011-04-26 03:13:18             341           170             3555   
178988  2010-09-29 00:07:38            4296          2922            27184   
179037  2009-05-29 16:48:39              96           454           124175   
179092  2015-07-02 23:07:03              68           192             2455   
179097  2017-08-16 22:12:17             709          1158            95006   

        user_verified                 date  \
7               False  2020-07-25 12:27:00   
10              False  2020-07-25 12:26:53   
18              False  2020-07-25 12:26:43   
20              False  2020-07-25 12:26:37   
30              False  2020-07-25 12:26:21   
...               ...                  ...   
178978          False  2020-08-29 19:51:22   
178988          False  2020-08-29 19:50:37   
179037          False  2020-08-29 19:48:10   
179092          False  2020-08-29 19:44:49   
179097          False  2020-08-29 19:44:35   

                                                     text  \
7       You now have to wear face coverings when out s...   
10      49K+ Covid19 cases still no response from \n@c...   
18      It is during our darkest moments that we must ...   
20      @EvanAKilgore @realDonaldTrump Good Patriots!\...   
30      #FEMA acknowledges #PuertoRico lacks rebuilt h...   
...                                                   ...   
178978  My first week back in school has taught me two...   
178988  7 consecutive days without a death from #Covid...   
179037  Trump Program to Cover Uninsured Covid-19 Pati...   
179092  @politvidchannel The Trump Administration's In...   
179097  @politvidchannel Just not the lives of #COVID1...   

                                        hashtags               source  \
7                                            NaN            TweetDeck   
10                                           NaN      Twitter Web App   
18                                     ['light']      Twitter Web App   
20                                           NaN   Twitter for iPhone   
30             ['FEMA', 'PuertoRico', 'COVID19']   Twitter for iPhone   
...                                          ...                  ...   
178978                                       NaN   Twitter for iPhone   
178988                               ['Covid19']  Twitter for Android   
179037   ['HealthInsurance', 'TrumpIsKillingUs']     Twitter for iPad   
179092                                 ['COVID']      Twitter Web App   
179097  ['COVID19', 'coronavirus', 'TrumpVirus']  Twitter for Android   

        is_retweet  
7            False  
10           False  
18           False  
20           False  
30           False  
..

In [6]:
df.shape

(179108, 13)

o DataFrame possui no total 179108 linhas, se excluirmos todos os dados NaN, de ambas as colunas, estariamos excluindo um maximo de 47059 (desconsiderando que algumas linhas possam ter NaN em ambas colunas).

# Excluindo dados

In [9]:
df_exclui = df.dropna(subset=["user_description", "user_location"])

In [10]:
df_exclui.shape

(138352, 13)

In [11]:
df_exclui.loc[df_exclui["user_description"].isna()]

Empty DataFrame
Columns: [user_name, user_location, user_description, user_created, user_followers, user_friends, user_favourites, user_verified, date, text, hashtags, source, is_retweet]
Index: []

In [12]:
df_exclui.loc[df_exclui["user_location"].isna()]

Empty DataFrame
Columns: [user_name, user_location, user_description, user_created, user_followers, user_friends, user_favourites, user_verified, date, text, hashtags, source, is_retweet]
Index: []

In [15]:
df_exclui.loc[df_exclui["hashtags"].isna()]

user_name               user_location  \
0                  ᏉᎥ☻լꂅϮ                  astroworld   
1           Tom Basile 🇺🇸                New York, NY   
14         DailyaddaaNews                   New Delhi   
19        Coronavirus Law                Florida, USA   
21      APO Group English               #AFRICA #MENA   
...                   ...                         ...   
179085     3D Innovations            Honolulu, Hawaii   
179090   Randy Richardson              Des Moines, IA   
179096     laura l. sweet             Los Angeles, CA   
179105        BEEHEMOTH ⏳                   🇨🇦 Canada   
179107            TUKY II  Aliwal North, South Africa   

                                         user_description  \
0       wednesday addams as a disney princess keepin i...   
1       Husband, Father, Columnist & Commentator. Auth...   
14                       Breaking news alerts from India.   
19                       COVID-19 Practice of Lechner Law   
21      Latest #Africa & #MENA related #News releases ...   
...                                                   ...   
179085  Providing Engineering/3D Design, 3D Printing, ...   
179090  Organizer with Iowans for Public Ed. Former (a...   
179096  Creative Director/AD/CW, Advertising & Brandin...   
179105  ⚒️ The Architects of Free Trade ⚒️ Really Did ...   
179107  TOKELO SEKHOPA | TUKY II | LAST BORN | EISH TU...   

               user_created  user_followers  user_friends  user_favourites  \
0       2017-05-26 05:46:42             624           950            18775   
1       2009-04-16 20:06:23            2253          1677               24   
14      2016-10-22 09:18:42             546            29               88   
19      2019-12-03 19:00:11              14            24               74   
21      2011-02-22 09:09:45           10661             6             2037   
...                     ...             ...           ...              ...   
179085  2011-07-06 19:50:05            5776          3762            13554   
179090  2009-04-13 20:33:21            2873          4846            16713   
179096  2008-03-18 16:45:55            2146          1077             5306   
179105  2016-07-13 17:21:59            1623          2160            98000   
179107  2018-04-14 17:30:07              97          1697              566   

        user_verified                 date  \
0               False  2020-07-25 12:27:21   
1                True  2020-07-25 12:27:17   
14              False  2020-07-25 12:26:46   
19              False  2020-07-25 12:26:39   
21               True  2020-07-25 12:26:31   
...               ...                  ...   
179085          False  2020-08-29 19:45:01   
179090          False  2020-08-29 19:44:53   
179096          False  2020-08-29 19:44:40   
179105          False  2020-08-29 19:44:15   
179107          False  2020-08-29 19:44:08   

                                                     text hashtags  \
0       If I smelled the scent of hand sanitizers toda...      NaN   
1       Hey @Yankees @YankeesPR and @MLB - wouldn't it...      NaN   
14      Rajasthan Government today started a Plasma Ba...      NaN   
19      COVID Update: The infection rate in Florida is...      NaN   
21      Coronavirus - South Africa: COVID-19 update fo...      NaN   
...                                                   ...      ...   
179085  How toilet paper and avocados help explain the...      NaN   
179090  Johnston school district currently has 4 stude...      NaN   
179096  We were really bummed we couldn’t cop one of t...      NaN   
179105  @CTVNews A powerful painting by Juan Lucena. I...      NaN   
179107  I stop when I see a Stop\n\n@SABCNews\n@Izinda...      NaN   

                     source  is_retweet  
0        Twitter for iPhone       False  
1       Twitter for Android       False  
14          Twitter Web App       False  
19         Twitter for iPad       False  
21          Africa Newsroom       False  
...            

# Filtrando dados com valor

In [16]:
df["user_location"] = df["user_location"].fillna("")

In [12]:
df

user_name               user_location  \
0                       ᏉᎥ☻լꂅϮ                  astroworld   
1                Tom Basile 🇺🇸                New York, NY   
2              Time4fisticuffs            Pewee Valley, KY   
3                  ethel mertz        Stuck in the Middle    
4                     DIPR-J&K           Jammu and Kashmir   
...                        ...                         ...   
179103  AJIMATI AbdulRahman O.             Ilorin, Nigeria   
179104                   Jason                     Ontario   
179105             BEEHEMOTH ⏳                   🇨🇦 Canada   
179106           Gary DelPonte               New York City   
179107                 TUKY II  Aliwal North, South Africa   

                                         user_description  \
0       wednesday addams as a disney princess keepin i...   
1       Husband, Father, Columnist & Commentator. Auth...   
2       #Christian #Catholic #Conservative #Reagan #Re...   
3       #Browns #Indians #ClevelandProud #[]_[] #Cavs ...   
4       🖊️Official Twitter handle of Department of Inf...   
...                                                   ...   
179103    Animal Scientist|| Muslim|| Real Madrid/Chelsea   
179104  When your cat has more baking soda than Ninja ...   
179105  ⚒️ The Architects of Free Trade ⚒️ Really Did ...   
179106  Global UX UI Visual Designer. StoryTeller, Mus...   
179107  TOKELO SEKHOPA | TUKY II | LAST BORN | EISH TU...   

               user_created  user_followers  user_friends  user_favourites  \
0       2017-05-26 05:46:42             624           950            18775   
1       2009-04-16 20:06:23            2253          1677               24   
2       2009-02-28 18:57:41            9275          9525             7254   
3       2019-03-07 01:45:06             197           987             1488   
4       2017-02-12 06:45:15          101009           168              101   
...                     ...             ...           ...              ...   
179103  2013-12-30 18:59:19             412          1609             1062   
179104  2011-12-21 04:41:30             150           182             7295   
179105  2016-07-13 17:21:59            1623          2160            98000   
179106  2009-10-27 17:43:13            1338          1111                0   
179107  2018-04-14 17:30:07              97          1697              566   

        user_verified                 date  \
0               False  2020-07-25 12:27:21   
1                True  2020-07-25 12:27:17   
2               False  2020-07-25 12:27:14   
3               False  2020-07-25 12:27:10   
4               False  2020-07-25 12:27:08   
...               ...                  ...   
179103          False  2020-08-29 19:44:21   
179104          False  2020-08-29 19:44:16   
179105          False  2020-08-29 19:44:15   
179106          False  2020-08-29 19:44:14   
179107          False  2020-08-29 19:44:08   

                                                     text  \
0       If I smelled the scent of hand sanitizers toda...   
1       Hey @Yankees @YankeesPR and @MLB - wouldn't it...   
2       @diane3443 @wdunlap @realDonaldTrump Trump nev...   
3       @brookbanktv The one gift #COVID19 has give me...   
4       25 July : Media Bulletin on Novel #CoronaVirus...   
...                                                   ...   
179103  Thanks @IamOhmai for nominating me for the @WH...   
179104  2020! The year of insanity! Lol! #COVID19 http...   
179105  @CTVNews A powerful painting by Juan Lucena. I...   
179106  More than 1,200 students test positive for #CO...   
179107  I stop when I see a Stop\n\n@SABCNews\n@Izinda...   

                                 hashtags               source  is_retweet  
0                                     NaN   Twitter for iPhone       False  
1                                     NaN  Twitter for Android       False  
2                             ['COVID19']  Twitter for Android       False  
3           

In [13]:
df.loc[df["user_location"].isna()]

Empty DataFrame
Columns: [user_name, user_location, user_description, user_created, user_followers, user_friends, user_favourites, user_verified, date, text, hashtags, source, is_retweet]
Index: []

In [14]:
df.loc[df["user_location"] == ""]

user_name user_location  \
7               Derbyshire LPC                 
8            Prathamesh Bendre                 
10      Voice Of CBSE Students                 
16              ChennaiCityNow                 
18                  Dorian Aur                 
...                        ...           ...   
179083           Auntie Kittie                 
179091             🦋  ROGUE  🦋                 
179092       claudia fernandez                 
179097               rainmaker                 
179098               John Geer                 

                                         user_description  \
7                                                     NaN   
8        A poet, reiki practitioner and a student of law.   
10                                                    NaN   
16      Individual tweeting about significant happenin...   
18                                                    NaN   
...                                                   ...   
179083  I'm the auntie you always wished you had.  If ...   
179091  May all aching hearts find their way there thr...   
179092                                                NaN   
179097                                                NaN   
179098  #StayAtHome #StayAtHomeSaveLifes #MaskUp \nFor...   

               user_created  user_followers  user_friends  user_favourites  \
7       2012-02-03 18:08:10             608           355               95   
8       2015-04-25 08:15:41              25            29               18   
10      2020-07-14 17:50:30               8            10                7   
16      2009-04-26 09:38:11            3987            53              749   
18      2011-01-30 18:40:16              46           108              453   
...                     ...             ...           ...              ...   
179083  2019-12-25 18:36:22             888           545               33   
179091  2017-01-30 07:38:23             628           386            14845   
179092  2015-07-02 23:07:03              68           192             2455   
179097  2017-08-16 22:12:17             709          1158            95006   
179098  2020-04-18 01:55:14              61           168            10817   

        user_verified                 date  \
7               False  2020-07-25 12:27:00   
8               False  2020-07-25 12:26:59   
10              False  2020-07-25 12:26:53   
16              False  2020-07-25 12:26:44   
18              False  2020-07-25 12:26:43   
...               ...                  ...   
179083          False  2020-08-29 19:45:02   
179091          False  2020-08-29 19:44:51   
179092          False  2020-08-29 19:44:49   
179097          False  2020-08-29 19:44:35   
179098          False  2020-08-29 19:44:34   

                                                     text  \
7       You now have to wear face coverings when out s...   
8       Praying for good health and recovery of @Chouh...   
10      49K+ Covid19 cases still no response from \n@c...   
16      July 25 #COVID19 update\n#TamilNadu - 6988\nDi...   
18      It is during our darkest moments that we must ...   
...                                                   ...   
179083  Just a quick note for my lovely members.  Soci...   
179091  @dougmillsnyt @realDonaldTrump @FLOTUS "The #T...   
179092  @politvidchannel The Trump Administration's In...   
179097  @politvidchannel Just not the lives of #COVID1...   
179098  Report #COVID19 outbreaks in K-12 schools here...   

                                                 hashtags  \
7                                                     NaN   
8                            ['covid19', 'covidPositive']   
10                                                    NaN   
16                    ['COVID19', 'TamilNadu', 'chennai']   
18                                              ['light']   
...                                                   ...   
179083                                                NaN   


In [15]:
df.loc[df["user_description"].isna()]

user_name  user_location user_description  \
7               Derbyshire LPC                             NaN   
10      Voice Of CBSE Students                             NaN   
18                  Dorian Aur                             NaN   
20           The Voice of GenX                             NaN   
30                      Ms Paz  United States              NaN   
...                        ...            ...              ...   
178978               Cody Vest                             NaN   
178988                   Shane        Ireland              NaN   
179037            Carol London                             NaN   
179092       claudia fernandez                             NaN   
179097               rainmaker                             NaN   

               user_created  user_followers  user_friends  user_favourites  \
7       2012-02-03 18:08:10             608           355               95   
10      2020-07-14 17:50:30               8            10                7   
18      2011-01-30 18:40:16              46           108              453   
20      2011-05-28 15:28:21             292          1037               58   
30      2019-09-15 18:10:09             127           974            30217   
...                     ...             ...           ...              ...   
178978  2011-04-26 03:13:18             341           170             3555   
178988  2010-09-29 00:07:38            4296          2922            27184   
179037  2009-05-29 16:48:39              96           454           124175   
179092  2015-07-02 23:07:03              68           192             2455   
179097  2017-08-16 22:12:17             709          1158            95006   

        user_verified                 date  \
7               False  2020-07-25 12:27:00   
10              False  2020-07-25 12:26:53   
18              False  2020-07-25 12:26:43   
20              False  2020-07-25 12:26:37   
30              False  2020-07-25 12:26:21   
...               ...                  ...   
178978          False  2020-08-29 19:51:22   
178988          False  2020-08-29 19:50:37   
179037          False  2020-08-29 19:48:10   
179092          False  2020-08-29 19:44:49   
179097          False  2020-08-29 19:44:35   

                                                     text  \
7       You now have to wear face coverings when out s...   
10      49K+ Covid19 cases still no response from \n@c...   
18      It is during our darkest moments that we must ...   
20      @EvanAKilgore @realDonaldTrump Good Patriots!\...   
30      #FEMA acknowledges #PuertoRico lacks rebuilt h...   
...                                                   ...   
178978  My first week back in school has taught me two...   
178988  7 consecutive days without a death from #Covid...   
179037  Trump Program to Cover Uninsured Covid-19 Pati...   
179092  @politvidchannel The Trump Administration's In...   
179097  @politvidchannel Just not the lives of #COVID1...   

                                        hashtags               source  \
7                                            NaN            TweetDeck   
10                                           NaN      Twitter Web App   
18                                     ['light']      Twitter Web App   
20                                           NaN   Twitter for iPhone   
30             ['FEMA', 'PuertoRico', 'COVID19']   Twitter for iPhone   
...                                          ...                  ...   
178978                                       NaN   Twitter for iPhone   
178988                               ['Covid19']  Twitter for Android   
179037   ['HealthInsurance', 'TrumpIsKillingUs']     Twitter for iPad   
179092                                 ['COVID']      Twitter Web App   
179097  ['COVID19', 'coronavirus', 'TrumpVirus']  Twitter for Android   

        is_retweet  
7            False  
10           False  
18           False  
20           False  
30           False  
..

In [17]:
# podemos usar qualquer valor como filtro

df["user_description"] = df["user_description"].fillna("oi")

In [19]:
df_depois = df.loc[df["user_description"].str.contains("oi")]

In [20]:
df_depois

user_name      user_location  \
7                          Derbyshire LPC                      
10                 Voice Of CBSE Students                      
18                             Dorian Aur                      
20                      The Voice of GenX                      
27             beatnikgeek the soothsayer      Manhattan, NY   
...                                   ...                ...   
179058                          Don Stand  Where my heart is   
179061  EpiTrace© Sentinel Solutions Inc.          Canada 🇨🇦   
179063               IL State Med Society                      
179092                  claudia fernandez                      
179097                          rainmaker                      

                                         user_description  \
7                                                      oi   
10                                                     oi   
18                                                     oi   
20                                                     oi   
27      These days, I expose colonizers & exploits @ t...   
...                                                   ...   
179058  Dad, husband, friend | Aim 2b a good human bei...   
179061  Real-time pandemic early warning system design...   
179063  Illinois State Medical Society: doctors workin...   
179092                                                 oi   
179097                                                 oi   

               user_created  user_followers  user_friends  user_favourites  \
7       2012-02-03 18:08:10             608           355               95   
10      2020-07-14 17:50:30               8            10                7   
18      2011-01-30 18:40:16              46           108              453   
20      2011-05-28 15:28:21             292          1037               58   
27      2008-02-23 19:02:29              86           259             9412   
...                     ...             ...           ...              ...   
179058  2009-07-05 21:12:55             137           277            12013   
179061  2020-08-26 02:51:36              36           756               51   
179063  2008-10-29 18:31:28            2758           798              744   
179092  2015-07-02 23:07:03              68           192             2455   
179097  2017-08-16 22:12:17             709          1158            95006   

        user_verified                 date  \
7               False  2020-07-25 12:27:00   
10              False  2020-07-25 12:26:53   
18              False  2020-07-25 12:26:43   
20              False  2020-07-25 12:26:37   
27              False  2020-07-25 12:26:26   
...               ...                  ...   
179058          False  2020-08-29 19:46:30   
179061          False  2020-08-29 19:46:07   
179063          False  2020-08-29 19:46:01   
179092          False  2020-08-29 19:44:49   
179097          False  2020-08-29 19:44:35   

                                                     text  \
7       You now have to wear face coverings when out s...   
10      49K+ Covid19 cases still no response from \n@c...   
18      It is during our darkest moments that we must ...   
20      @EvanAKilgore @realDonaldTrump Good Patriots!\...   
27      I can imagine the same people profiting off th...   
...                                                   ...   
179058  Here is the unhinged older brother of antisemi...   
179061  COVID-19 outbreak: gold rush and responsibilit...   
179063  .@Illinois_Alma’s #COVID19 testing plan is so ...   
179092  @politvidchannel The Trump Administration's In...   
179097  @politvidchannel Just not the lives of #COVID1...   

                                        hashtags               source  \
7                                            NaN            TweetDeck   
10                                           NaN      Twitter Web App   
18                                     ['light']      Twitter Web App   
20             

In [21]:
df

user_name               user_location  \
0                       ᏉᎥ☻լꂅϮ                  astroworld   
1                Tom Basile 🇺🇸                New York, NY   
2              Time4fisticuffs            Pewee Valley, KY   
3                  ethel mertz        Stuck in the Middle    
4                     DIPR-J&K           Jammu and Kashmir   
...                        ...                         ...   
179103  AJIMATI AbdulRahman O.             Ilorin, Nigeria   
179104                   Jason                     Ontario   
179105             BEEHEMOTH ⏳                   🇨🇦 Canada   
179106           Gary DelPonte               New York City   
179107                 TUKY II  Aliwal North, South Africa   

                                         user_description  \
0       wednesday addams as a disney princess keepin i...   
1       Husband, Father, Columnist & Commentator. Auth...   
2       #Christian #Catholic #Conservative #Reagan #Re...   
3       #Browns #Indians #ClevelandProud #[]_[] #Cavs ...   
4       🖊️Official Twitter handle of Department of Inf...   
...                                                   ...   
179103    Animal Scientist|| Muslim|| Real Madrid/Chelsea   
179104  When your cat has more baking soda than Ninja ...   
179105  ⚒️ The Architects of Free Trade ⚒️ Really Did ...   
179106  Global UX UI Visual Designer. StoryTeller, Mus...   
179107  TOKELO SEKHOPA | TUKY II | LAST BORN | EISH TU...   

               user_created  user_followers  user_friends  user_favourites  \
0       2017-05-26 05:46:42             624           950            18775   
1       2009-04-16 20:06:23            2253          1677               24   
2       2009-02-28 18:57:41            9275          9525             7254   
3       2019-03-07 01:45:06             197           987             1488   
4       2017-02-12 06:45:15          101009           168              101   
...                     ...             ...           ...              ...   
179103  2013-12-30 18:59:19             412          1609             1062   
179104  2011-12-21 04:41:30             150           182             7295   
179105  2016-07-13 17:21:59            1623          2160            98000   
179106  2009-10-27 17:43:13            1338          1111                0   
179107  2018-04-14 17:30:07              97          1697              566   

        user_verified                 date  \
0               False  2020-07-25 12:27:21   
1                True  2020-07-25 12:27:17   
2               False  2020-07-25 12:27:14   
3               False  2020-07-25 12:27:10   
4               False  2020-07-25 12:27:08   
...               ...                  ...   
179103          False  2020-08-29 19:44:21   
179104          False  2020-08-29 19:44:16   
179105          False  2020-08-29 19:44:15   
179106          False  2020-08-29 19:44:14   
179107          False  2020-08-29 19:44:08   

                                                     text  \
0       If I smelled the scent of hand sanitizers toda...   
1       Hey @Yankees @YankeesPR and @MLB - wouldn't it...   
2       @diane3443 @wdunlap @realDonaldTrump Trump nev...   
3       @brookbanktv The one gift #COVID19 has give me...   
4       25 July : Media Bulletin on Novel #CoronaVirus...   
...                                                   ...   
179103  Thanks @IamOhmai for nominating me for the @WH...   
179104  2020! The year of insanity! Lol! #COVID19 http...   
179105  @CTVNews A powerful painting by Juan Lucena. I...   
179106  More than 1,200 students test positive for #CO...   
179107  I stop when I see a Stop\n\n@SABCNews\n@Izinda...   

                                 hashtags               source  is_retweet  
0                                     NaN   Twitter for iPhone       False  
1                                     NaN  Twitter for Android       False  
2                             ['COVID19']  Twitter for Android       False  
3           